File: DataStructure/74. Search a 2D Matrix.cs
public class Solution {
    public bool SearchMatrix(int[][] matrix, int target) {
        foreach (int[] m in matrix)
        {
            List<int> newM = m.ToList();
            if (newM.BinarySearch(target) >= 0)
                return true;
        }
        return false;
    }
}

File: LeetCode/DataStructure/74. Search a 2D Matrix.py
class Solution:    
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        def binary_search(array, target):
            array.sort() # 정렬
            left = 0
            right = len(array) - 1

            while left <= right:
                mid = (left + right) // 2 
                if array[mid] == target: 
                    return mid 
                elif array[mid] > target: 
                    right = mid - 1 
                else: 
                    left = mid + 1
                    
            return -1 
        
        for m in matrix:
            if binary_search(m, target) != -1:
                return True
        return False